runtime/update_runtime.sh

```diff
@@ -6,41 +6,106 @@
 # Default: Build is the one specified in file configure.ac
 # located at the root of the repository.
 set -x
-AUTHOR=${AUTHOR:-$(git config user.name)}
-AUTHOR_EMAIL=${AUTHOR_EMAIL:-$(git config user.email)}
-
-source ../versions.txt
-VERSION=$cc_runtime_version
-
-# If we are providing the branch or hash to build, assign it to OBS_REVISION
-if [ -n "$1" ]; then
-     OBS_REVISION=$1
-
-     # Validate input is alphanumeric, commit ID
-     # If a commit ID is provided, override versions.txt one
-     if [[ "$OBS_REVISION" =~ ^[a-zA-Z0-9][-a-zA-Z0-9]{0,40}[a-zA-Z0-9]$  ]]; then
-         hash_tag=$OBS_REVISION
-     else
-         hash_tag=$cc_runtime_hash
-     fi
-else
-         hash_tag=$cc_runtime_hash
-fi
-short_hashtag="${hash_tag:0:7}"
-
-OBS_PUSH=${OBS_PUSH:-false}
-OBS_RUNTIME_REPO=${OBS_RUNTIME_REPO:-home:clearcontainers:clear-containers-3-staging/cc-runtime}
-OBS_APIURL=${OBS_APIURL:-""}
-
-# This allows to point to internal/private OBS instance
-if [ "$OBS_APIURL" != "" ]; then
-    APIURL="-A ${OBS_APIURL}"
-else
-    APIURL=""
-fi
-
-echo "Running: $0 $@"
-echo "Update cc-runtime $VERSION: ${hash_tag:0:7}"
+
+SCRIPT_NAME=$0
+SCRIPT_DIR=$(dirname $0)
+
+COMMIT=false
+BRANCH=false
+LOCAL_BUILD=false
+OBS_PUSH=false
+VERBOSE=false
+
+BUILD_DISTROS=(Fedora_26 xUbuntu_16.04)
+
+function display_help()
+{
+  cat <<-EOL 
+  $SCRIPT_NAME
+
+  This script is intended to create Clear Containers 3.0 packages for the OBS 
+  (Open Build Service) platform.
+
+  Usage:
+        $SCRIPT_NAME [options]
+
+  Options:
+
+    -l         --local-build     Build the runtime locally
+    -w	       --workdir	 Repository directory
+    -c         --commit-id       Build with a given commit ID
+    -b         --branch          Build with a given branch name
+    -p         --push            Push changes to OBS
+    -a         --api-url         Especify an OBS API (e.g. private OBS)
+    -r         --obs-repository  An OBS repository to push the changes.
+    -w         --workdir         Directory of a working copy of the OBS runtime repo
+    -v         --verbose         Set the -x flag for verbosity
+    -C         --clean           Clean the repository
+    -V         --verify          Verify the environment
+    -h         --help            Display this help message
+
+    Usage example
+
+    $SCRIPT_NAME --local-build --branch staging
+    $SCRIPT_NAME --commit-id a76f45c --push --api-url home:userx/runtime
+    $SCRIPT_NAME --commit-id a76f45c --push --obs-repository http://127.0.0.1
+    $SCRIPT_NAME --commit-id a76f45c --push
+
+EOL
+  exit 1
+}
+
+function verify()
+{
+    # This function perform some checks in order to make sure
+    # the script will run flawlessly.
+
+    # Make sure this script is called from ./
+    [ "$SCRIPT_DIR" != "." ] && echo "The script must be called from its base dir." && exit 1 || :
```

Comment:
For clarity better to use: 

`[ "$SCRIPT_DIR" != "." ] && echo "The script must be called from its base dir." && exit 1 || true`